pkg/storage/ingest/reader.go

```diff
@@ -35,6 +35,7 @@ const (
 )
 
 type record struct {
+	ctx      context.Context
```

Comment:
Can you add a comment to explain what does this context contain, and what it is used for, please?